### ISSUE TITLE
Cannot run ticgitweb on Ubuntu 10.10

### ISSUE THREAD

**Issue by alecthegeek** [jeffWelling/ticgit#19]
After installing ticgit-ng as per the readme and being able to run ti successfully I am unable to run the web interface:

$ ticgitweb 
You need to install ticgit before we can proceed
/var/lib/gems/1.8/gems/TicGit-ng-1.0.2.2/bin/ticgitweb:27: uninitialized constant TicGit (NameError)

**Comment by jeffWelling** [comment]
Ah.  This looks familiar.  
I thought this had been patched already... At least, this specific problem. 

 I can't say for sure if ticgitweb would run or not because I don't use it myself, it's kind of a legacy app from previous maintainers, but ticgit itself hasn't been changed in a way that should effect ticgitweb so.. _shrugs_ I can't say for sure.  
I'll fix this, but before I sink into another bug I have to do some chores around the house, shouldn't take an hour.  
I'll bring up that VM from the other issue and see if I can poke and prod ticgitweb until it works.  
The documentation should also be mostly accurate, there was a maintainer working with me not to long ago (CodeGnome) who I believe had ticgitweb in a functioning state, and had updated the appropriate documentation accordingly.  

I'll post back with the results of my poking and prodding :)

**Comment by jeffWelling** [comment]
I believe this should be fixed with TicGit-ng version 1.0.2.3, just pushed to rubygems, rubyforge, and Github now.
Try it out and let me know if it works for you?
After making this small change, I was able to run `ticgitweb` with one command.

**Comment by jeffWelling** [comment]
Oh no, perhaps I pushed too soon, trying it on my VM instead of my deving box, it seems to silently fail.
Does this happen for you as well? 
It seems like it may be a ubuntu-specific issue, as this isn't appearing on my deving box which is Debian.

**Comment by alecthegeek** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

It fails silently for me as well.

I also tried specifying a port of 2000 and it still failed.

---

Alec Clews
Personal alec.clews@gmail.com                  Melbourne, Australia
Jabber:  alecclews@jabber.org                   PGPKey ID: 0x9BBBFC7C
Blog  http://alecthegeek.wordpress.com/
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1.4.10 (GNU/Linux)
Comment: Using GnuPG with Mozilla - http://enigmail.mozdev.org/

iEYEARECAAYFAk2hBFsACgkQP+e5j5u7/Hz1mgCgkeE2BfBvR2a0tbjj1O6hsOkA
LGsAnR5E4gLkBgmfpQOHFz0hAGVS3lEn
=V15a
-----END PGP SIGNATURE-----

**Comment by jeffWelling** [comment]
Hmmm.
So, just for posterity I want to ask, how are you calling ticgitweb?
I ask because on my Ubuntu system I run into this bug  
https://github.com/jeffWelling/ticgit/issues/16

And I'm wondering if you are also seeing that bug?

Because if you are that means you have to call it (both ti and ticgitweb) from a full path, such as
`/var/lib/gems/1.8/bin/ticgitweb`  
Which is silently failing.  Now, here's the strange part, at least to me.  That file, isn't actually ticgitweb, it's some file that rubygems places there which I assume (based on the premise that the 'ti' file in that directory is also not the real 'ti' file, yet the system works) it's loading the real file.  Doing `locate bin/ticgitweb` results in

```
/home/jeff/ticgit/bin/ticgitweb
/var/lib/gems/1.8/bin/ticgitweb
/var/lib/gems/1.8/gems/TicGit-ng-1.0.2.2/bin/ticgitweb
/var/lib/gems/1.8/gems/TicGit-ng-1.0.2.3/bin/ticgitweb
```

So, if I call `/var/lib/gems/1.8/gems/TicGit-ng-1.0.2.3/bin/ticgitweb` then it starts up without problems on 0.0.0.0:4567 .
Why it's failing when you call the first one though, I have no idea.  I'm still investigating but my present unfamiliarity with sinatra and what is going on with that file have me thinking this could take a while to work out.

Checking my Debian vm, that strange file exists there as well, so I take it that this is a result of rubygems packaging as the file indicated.

So currently we know that this problem appears to be Ubuntu specific (and perhaps doesn't have anything to do with the mystery file), even though it _does_ work if you call `ticgitweb` by its absolute path.

I'm going to have to sleep on it, and perhaps post to programmers.stackexchange.com to try to get help with this one.  Will post back when I find out more.  For the time being as a work-around you can add this alias to your `~/.bashrc` file.  
    alias ticgitweb=/var/lib/gems/1.8/gems/TicGit-ng-1.0.2.3/bin/ticgitweb
With that alias, calling `ticgitweb` results in a running server on my Ubuntu VM.
...*reaches for a larger banging utensil*

**Comment by alecthegeek** [comment]
It's not working that well for me

```
$ locate -br '^ticgitweb$'
/var/lib/gems/1.8/bin/ticgitweb
/var/lib/gems/1.8/gems/TicGit-ng-1.0.2.2/bin/ticgitweb

$/var/lib/gems/1.8/gems/TicGit-ng-1.0.2.2/bin/ticgitweb
You need to install ticgit before we can proceed
```

**Comment by jeffWelling** [comment]
Can you try installing / updating to version 1.0.2.3?
    $ sudo gem update
I think.
And then for posterity
    $ sudo updatedb
And then try the locate again.  Mine showed version 1.0.2.2 and version 1.0.2.3 .

If that doesn't do it, we'll have to wait for someone to respond to the question on answers.launchpad.net .  I'm still trying to figure out what to do about that "You need to install ticgit before we can proceed" message.

**Comment by alecthegeek** [comment]
Success:

1) gem update did not work so I just did an install
2) I added /var/lib/gems/1.8/gems/TicGit-ng-1.0.2.3/bin to the PATH. I suspect this was significant

Now runs

w00t!

**Comment by alecthegeek** [comment]
For the benefit of posterity. As gem update does not work you can remove the previous version by running

```
sudo gem uninstall TicGit-ng -v 1.0.2.2
```

**Comment by jeffWelling** [comment]
Awesome!

Gem update didn't work? Hmm. That might be something to look into. Is
it just for Ticgit that it's not working or is it all gems (not my
problem)? Lol

Yeah the path thing seems to be the crux of this whole problem, but I
don't yet know why rubygems isn't adding that to the path when you
install it.
Still looking into it, but I'm glad you got it working :)

On 2011-04-10, at 6:14 PM, alecthegeek
reply@reply.github.com
wrote:

**Comment by alecthegeek** [comment]
Only the TicGit Gem did not upgrade. Sinatra upgraded fine....

**Comment by jeffWelling** [comment]
How did you originally install ticgit? Did you download and install
the gem from github or rubyforge?
Were there any error messages indicating what failed?

On 2011-04-10, at 6:59 PM, alecthegeek
reply@reply.github.com
wrote:

**Comment by alecthegeek** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

On 11/04/11 09:11, jeffWelling wrote:

> How did you originally install ticgit? Did you download and install
> the gem from github or rubyforge?

I installed from the command line as per the README

```
sudo gem install ......
```

Also note that ticgit has _not_ been installed from the Ubuntu repos

> Were there any error messages indicating what failed?

AFAIK the only problem was installing sass gem, which needed the --pre flag

---

Alec Clews
Personal alec.clews@gmail.com                  Melbourne, Australia
Jabber:  alecclews@jabber.org                   PGPKey ID: 0x9BBBFC7C
Blog  http://alecthegeek.wordpress.com/
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1.4.10 (GNU/Linux)
Comment: Using GnuPG with Mozilla - http://enigmail.mozdev.org/

iEYEARECAAYFAk2iSnUACgkQP+e5j5u7/Hy5pQCeIB7QsVobLHvoKXhqzVwurAbX
syEAnRg0jUoObVKoh0oPVqnu3/Rn8NDv
=mQLW
-----END PGP SIGNATURE-----

**Comment by jeffWelling** [comment]
Oh sorry, I meant were there any error messages when you tried to update the TicGit-ng gem but it failed?

**Comment by alecthegeek** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

No -- just said nothing to update

On 11/04/11 11:00, jeffWelling wrote:

> Oh sorry, I meant were there any error messages when you tried to update the TicGit-ng gem but it failed?

---

Alec Clews
Personal alec.clews@gmail.com                  Melbourne, Australia
Jabber:  alecclews@jabber.org                   PGPKey ID: 0x9BBBFC7C
Blog  http://alecthegeek.wordpress.com/
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1.4.10 (GNU/Linux)
Comment: Using GnuPG with Mozilla - http://enigmail.mozdev.org/

iEYEARECAAYFAk2iVhAACgkQP+e5j5u7/HxSpgCghOdqMNpIpuD6seDeYoULhfCL
w2sAnjdfPbq4/CJLpFV+hAKKCQziC5+v
=0pUB
-----END PGP SIGNATURE-----

**Comment by jeffWelling** [comment]
Does this comment still stand then?
https://github.com/jeffWelling/ticgit/issues/19#issuecomment-979744

Does the output still only list 1.0.2.2?

lol Lets make this easier, what's the output of `ti --version`? If it says 1.0.2.3 then I'm chasing my tail lol

**Comment by alecthegeek** [comment]
```
$ ti --version
1.0.2.3
```

**Comment by jeffWelling** [comment]
So it turns out the solution is to manually add the rubygems bin dir to your path. This will be noted in the documentation.

**Comment by alecthegeek** [comment]
See commit d6741103715df0c85a20

**Comment by jeffWelling** [comment]
The problem of not being able to simply call `ticgitweb` has been resolved completely for (*crosses fingers*) all platforms.  There was a misconfiguration in sinatra that was preventing it from starting _because_ it wasn't being called directly.  Thanks to the friendly folks at the sinatra google group for pointing this out to me.

**Comment by jeffWelling** [comment]
*resolved completely in version 1.0.2.4 for all platforms

**Comment by alecthegeek** [comment]
Tested and working.

Gem also upgrades (but 1.0.2.3 is not removed)
And correctly issues an error message if not located in a git repo
